homeassistant/components/amberelectric/sensor.py

```diff
@@ -0,0 +1,591 @@
+        data_service: AmberDataService,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Sensor."""
+        super().__init__(coordinator)
+        self._site_id = site_id
+        self._channel_type = channel_type
+        self._platform_name = platform_name
+        self._data_service = data_service
+
+    @property
+    def name(self) -> str | None:
+        """Return the friendly name of the sensor."""
+        return (
+            self._platform_name
+            + " - "
+            + friendly_channel_type(self._channel_type)
+            + " "
+            + "Price"
+        )
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique id for each sensors."""
+        return slugify(
+            self._site_id + " " + friendly_channel_type(self._channel_type) + " Price"
+        )
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        if self._channel_type == ChannelType.FEED_IN:
+            return "mdi:solar-power"
+        if self._channel_type == ChannelType.CONTROLLED_LOAD:
+            return "mdi:clock-outline"
+        return "mdi:transmission-tower"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the sensors unit of measurement."""
+        return "¢/kWh"
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the current price in c/kWh."""
+        channel = self._data_service.current_prices.get(self._channel_type)
+        if channel:
+            if self._channel_type == ChannelType.FEED_IN:
+                return round(channel.per_kwh, 0) * -1
+            return round(channel.per_kwh, 0)
+        return None
+
+    @property
+    def device_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return additional pieces of information about the price."""
+        meta = self._data_service.current_prices.get(self._channel_type)
+        data = {}
+        if meta is not None:
+            data["duration"] = meta.duration
+            data["date"] = meta.date.isoformat()
+            data["per_kwh"] = round(meta.per_kwh)
+            if self._channel_type == ChannelType.FEED_IN:
+                data["per_kwh"] = data["per_kwh"] * -1
+            data["nem_date"] = meta.nem_time.isoformat()
+            data["spot_per_kwh"] = round(meta.spot_per_kwh)
+            data["start_time"] = meta.start_time.isoformat()
+            data["end_time"] = meta.end_time.isoformat()
+            data["renewables"] = round(meta.renewables)
+            data["estimate"] = meta.estimate
+            data["spike_status"] = meta.spike_status.value
+            data["channel_type"] = meta.channel_type.value
+
+            if meta.range is not None:
+                data["range_min"] = meta.range.min
+                data["range_max"] = meta.range.max
+
+        data[ATTR_ATTRIBUTION] = ATTRIBUTION
+        return data
+
+
+class AmberEnergyPriceSensor(CoordinatorEntity, SensorEntity):
```

Comment:
Do you mean cost ?

Reply:
The first one matches what they see in our app, and how we talk about prices (c/kWh). The second one is compatible with the energy dashboard ($/kWh).

At the time I wrote this, there wasn't an easy way to convert it (outside of template). Though that may have changed?

Reply:
We should stick with providing the data in a way that is compatible with how Home Assistant represents the data and drop the other sensor.